### DISCUSSION TITLE
What firmware to use for "Zigbee CC2652P Coordinator v4 USB Dongle Stick Home Assistant!"

### DISCUSSION THREAD

**Discussion by MrEaoP** [Koenkk/zigbee2mqtt#18239] [Devices]
Hi,

I bought my zigbee controller on Ebay 1,5 years ago from a guy based in Kiev, the Ukraine.
Hardly surprising, given the current situation in the Ukraine, he is no longer selling the device on Ebay. 

I noted that with 1.32.0 new firmware is supported. Which of course triggers me to want to update the firmware.

Before I brick my zigbee controller, it would be nice to know which firmware to use.
All I know is that when i bought it is was called;

_"Zigbee CC2652P Coordinator v4 USB Dongle Stick Home Assistant!"_

Is there a command to query the zigbee controller more details, and thus know which firmware to use?

**Comment by lux73** [comment]
do `lsusb -v` via Shell and search & compare the Output from your Coordinator especially Name with this List below the Firmware Downloads -> https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator/Z-Stack_3.x.0/bin

**be aware! if you are not really really sure that's the correct Coordinator do not flash!!**

**Reply by lux73** [reply]
anyhow i would ask in the HomeAssistant Forums to see if anyone had a similar Adapter running ;-)

**Reply by MrEaoP** [reply]
Thanks for the advice - I'll try the homeassisant forums!
The "lsusb -v" didn't give me anything conclusive...
The device works fine as it is, so "if it ain't broke - don't fix it" might apply...
